document/core/exec/numerics.rst

```diff
@@ -1620,3 +1882,70 @@ Conversions
    \begin{array}{lll@{\qquad}l}
    \reinterpret_{t_1,t_2}(c) &=& \bits_{t_2}^{-1}(\bits_{t_1}(c)) \\
    \end{array}
+
+
+.. _op-sat_u:
+
+:math:`\satu_N(i)`
+.......................
+
+* If :math:`i` is greater than :math:`2^N-1`, return :math:`2^N-1`.
+
+* Else return :math:`i`.
+
+.. math::
+   \begin{array}{lll@{\qquad}l}
+   \satu_N(i) &=& 2^N-1 & (\iff i > 2^N-1)\\
+   \satu_N(i) &=& i & (\otherwise)
+   \end{array}
+
+
+.. _op-sat_s:
+
+:math:`\sats_N(i)`
+.......................
+
+* Let :math:`j` be the :ref:`signed interpretation <aux-signed>` of :math:`i` of size :math:`N`.
+
+* If :math:`j` is less than :math:`-2^{N-1}`, return :math:`-2^{N-1}`.
+
+* If :math:`j` is greater than :math:`2^N-1`, return :math:`2^N-1`.
+
+* Else return :math:`i`.
+
+.. math::
+   \begin{array}{lll@{\qquad}l}
+   \sats_N(i) &=& -2^{N-1} & (\iff \signed(i) < -2^{N-1})\\
+   \sats_N(i) &=& 2^N-1 & (\iff \signed(i) > 2^N-1)\\
+   \sats_N(i) &=& i & (\otherwise)
+   \end{array}
+
+
+.. _op-narrow_s:
+
+:math:`\narrows_{M,N}(i)`
```

Comment:
It looks like this convention of `i<operation>` is for the numeric operators. For these conversion, `extend` is just `extend`, not `iextend`. Should we keep the name as `narrow`?

Reply:
Sunds plausible, but if this has the same status as plain `extend`, then don't you need an `inarrow` wrapper as well, same as for extend? Otherwise, how is it picked up by the operator conventions?

Reply:
In [exec/instructions.rst](https://www.ngzhian.com/simd/core/exec/instructions.html#t-2-mathsf-x-n-mathsf-xref-syntax-instructions-syntax-instr-simd-mathsf-narrow-t-1-mathsf-x-m-xref-syntax-instructions-syntax-sx-mathit-sx) we don't use the operator conventions, we directly refer to `narrow` with the correct types from the instruction's name.

Reply:
Ah, I see. But that causes another issue that I missed before (sorry): there is a generic execution rule for shape.vbinop, and by the definition of vbinop, that should also apply to narrow. So there are two overlapping exec rules matching narrow. Similarly for vunop and widen.

So unlike what I said before, I think you'll need to remove those instructions from vunop and vbinop -- they don't fit the pattern, because unlike the others, they do not work lane-wise. (And consequently, they need separate typing rules.)

Reply:
Done! See a5165cb, I created a new production `\vwiden` that just groups `widen_low` and `widen_high`, that way I don't have to repeat this in the validation.
`narrow` also has its own validation rule now.